action/sealedenvelope.go

```diff
@@ -98,11 +98,20 @@ func (sealed *SealedEnvelope) LoadProto(pbAct *iotextypes.Action) error {
 	if err := elp.LoadProto(pbAct.GetCore()); err != nil {
 		return err
 	}
+	// populate pubkey and signature
+	srcPub, err := crypto.BytesToPublicKey(pbAct.GetSenderPubKey())
+	if err != nil {
+		return err
+	}
 	encoding := pbAct.GetEncoding()
 	switch encoding {
 	case iotextypes.Encoding_ETHEREUM_RLP:
 		// verify action type can support RLP-encoding
-		if _, err := actionToRLP(elp.Action()); err != nil {
+		tx, err := actionToRLP(elp.Action())
+		if err != nil {
+			return err
+		}
+		if _, err = rlpSignedHash(tx, config.EVMNetworkID(), pbAct.GetSignature()); err != nil {
```

Comment:
IMO, this is doing extra work so that you can use `panic`, which is weird when you don't have to use `panic`

what if there are more func calls later in the SealEnvelope code? we need to call those all in advance, so that `panic` can be used

it defeats the purpose of `panic` where you have no choice, but here we can return 0-hash, which will fail `Verify` gracefully